File: resize_screenshots.py
import glob
from PIL import Image

def process_screenshots():
    out_dir = "ios_screenshots"
    import os
    if not os.path.exists(out_dir):
        os.makedirs(out_dir)

    target_width = 1242
    target_height = 2688
    
    images = glob.glob('flutter_*.png')
    if not images:
        print("No flutter_*.png files found")
        return
        
    print(f"Found {len(images)} images.")
    for path in images[:10]: # Process up to 10 screenshots
        with Image.open(path) as img:
            print(f"Processing {path}, size: {img.size}")
            
            # Create a new image with the target dimensions, let's say with a white or custom color background.
            # We can also scale the original image properly to fit.
            new_img = Image.new("RGB", (target_width, target_height), (0, 0, 0)) # black background or maybe blur
            
            # Calculate aspect ratio preserving resize
            img_ratio = img.width / img.height
            target_ratio = target_width / target_height
            
            if img_ratio > target_ratio:
                # image is wider
                new_w = target_width
                new_h = int(new_w / img_ratio)
            else:
                # image is taller
                new_h = target_height
                new_w = int(new_h * img_ratio)
                
            resized = img.resize((new_w, new_h), Image.Resampling.LANCZOS)
            
            # Center the image
            offset_x = (target_width - new_w) // 2
            offset_y = (target_height - new_h) // 2
            
            new_img.paste(resized, (offset_x, offset_y))
            
            out_path = os.path.join(out_dir, path)
            new_img.save(out_path)
            print(f"Saved {out_path}")

if __name__ == '__main__':
    process_screenshots()

File: resize_screenshots.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() async {
  final outDir = Directory('ios_screenshots');
  if (!outDir.existsSync()) {
    outDir.createSync();
  }

  const int targetWidth = 1242;
  const int targetHeight = 2688;

  final dir = Directory.current;
  final entities = dir.listSync();
  final imagePaths = entities
      .whereType<File>()
      .where((file) => file.path.endsWith('.png') && file.path.contains('flutter_'))
      .toList();
      
  imagePaths.sort((a, b) => a.path.compareTo(b.path));

  if (imagePaths.isEmpty) {
    print('No flutter_*.png files found.');
    return;
  }

  print('Found ${imagePaths.length} screenshots. Processing first 10...');
  
  // process the first 10 flutter_XX.png files
  for (var file in imagePaths.take(10)) {
    print('Processing ${file.path}...');
    final imageBytes = file.readAsBytesSync();
    final originalImage = img.decodeImage(imageBytes);
    
    if (originalImage == null) {
      print('Could not read image ${file.path}');
      continue;
    }
    
    // Create new image with black background
    final newImage = img.Image(width: targetWidth, height: targetHeight, numChannels: 3);
    for (var p in newImage) {
      p.r = 0; p.g = 0; p.b = 0;
    }
    
    final double imgRatio = originalImage.width / originalImage.height;
    final double targetRatio = targetWidth / targetHeight;
    
    int newW = targetWidth;
    int newH = targetHeight;
    
    if (imgRatio > targetRatio) {
      // image is wider
      newW = targetWidth;
      newH = (newW / imgRatio).round();
    } else {
      // image is taller
      newH = targetHeight;
      newW = (newH * imgRatio).round();
    }
    
    final resizedImage = img.copyResize(originalImage, width: newW, height: newH, interpolation: img.Interpolation.linear);
    
    final int dstX = (targetWidth - newW) ~/ 2;
    final int dstY = (targetHeight - newH) ~/ 2;
    
    img.compositeImage(newImage, resizedImage, dstX: dstX, dstY: dstY);
    
    final outPath = '${outDir.path}/${file.uri.pathSegments.last}';
    final outFile = File(outPath);
    outFile.writeAsBytesSync(img.encodePng(newImage));
    print('Saved $outPath');
  }
}
